src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LogisticRegression.cs

```diff
@@ -15,21 +15,21 @@
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Trainers;
 
-[assembly: LoadableClass(LogisticRegressionBinaryTrainer.Summary, typeof(LogisticRegressionBinaryTrainer), typeof(LogisticRegressionBinaryTrainer.Options),
+[assembly: LoadableClass(LbfgsLogisticRegressionTrainer.Summary, typeof(LbfgsLogisticRegressionTrainer), typeof(LbfgsLogisticRegressionTrainer.Options),
     new[] { typeof(SignatureBinaryClassifierTrainer), typeof(SignatureTrainer), typeof(SignatureFeatureScorerTrainer) },
-    LogisticRegressionBinaryTrainer.UserNameValue,
-    LogisticRegressionBinaryTrainer.LoadNameValue,
-    LogisticRegressionBinaryTrainer.ShortName,
+    LbfgsLogisticRegressionTrainer.UserNameValue,
+    LbfgsLogisticRegressionTrainer.LoadNameValue,
+    LbfgsLogisticRegressionTrainer.ShortName,
     "logisticregressionwrapper")]
 
-[assembly: LoadableClass(typeof(void), typeof(LogisticRegressionBinaryTrainer), null, typeof(SignatureEntryPointModule), LogisticRegressionBinaryTrainer.LoadNameValue)]
+[assembly: LoadableClass(typeof(void), typeof(LbfgsLogisticRegressionTrainer), null, typeof(SignatureEntryPointModule), LbfgsLogisticRegressionTrainer.LoadNameValue)]
 
 namespace Microsoft.ML.Trainers
 {
 
     /// <include file='doc.xml' path='doc/members/member[@name="LBFGS"]/*' />
     /// <include file='doc.xml' path='docs/members/example[@name="LogisticRegressionBinaryClassifier"]/*' />
-    public sealed partial class LogisticRegressionBinaryTrainer : LbfgsTrainerBase<LogisticRegressionBinaryTrainer.Options,
+    public sealed partial class LbfgsLogisticRegressionTrainer : LbfgsTrainerBase<LbfgsLogisticRegressionTrainer.Options,
```

Comment:
>LbfgsLogisticRegressionTrainer [](start = 32, length = 30)

Can you call it `LbfgsBinaryTrainer`?

Reply:
It looks fine if we consider this trainer and its SDCA counterpart (they all solve LR). However, it will break the consistency between LBFGS trainers. Please take a look at Iteration 3.

---
In reply to: [267460936](https://github.com/dotnet/machinelearning/pull/3034#discussion_r267460936) [](ancestors = 267460936)

Reply:
@sfilipi has comment on your issue, and I share it as well.
Putting word `Regression` without `Binary`  in the trainer is confusing.
If I would saw name of this trainer I would assume it's part of regression task.
Which is not.


---
In reply to: [267475700](https://github.com/dotnet/machinelearning/pull/3034#discussion_r267475700) [](ancestors = 267475700,267460936)

Reply:
yeah, given the target user base, IMO should do a one off for this one, and keep binary in the name. 

(also this feels like a classic academic vs engineering solution of  a problem :D )

---
In reply to: [267477762](https://github.com/dotnet/machinelearning/pull/3034#discussion_r267477762) [](ancestors = 267477762,267475700,267460936)

Reply:
It will make the API looks like `mlContext.BinaryClassification.LbfgsBinary(...)` so I switch to SDCA-style name `LbfgsCalibrated`.

---
In reply to: [267481155](https://github.com/dotnet/machinelearning/pull/3034#discussion_r267481155) [](ancestors = 267481155,267477762,267475700,267460936)

Reply:
> 
> 
> yeah, given the target user base, IMO should do a one off for this one, and keep binary in the name.
> 
> (also this feels like a classic academic vs engineering solution of a problem :D )
> 
> Also, IMO not all details need to be in the name. IMO we don't need to distingush that this is Lbfgs logistic regression through renaming. It can be in the documentations. shorter names are better...

There are several trainers returning the same model types. Only this one is called `LogisticRegression` and this one has been outperformed by other algorithms. Thus, I want to append `Lbfgs` to emphasize that 'this is not the only trainer for logistic regression`.

> 
> In reply to: [267477762](https://github.com/dotnet/machinelearning/pull/3034#discussion_r267477762) [](ancestors = 267477762,267475700,267460936)